frontend/storage/utils/apiRequestManager.ts

```diff
@@ -0,0 +1,33 @@
+import axios from 'axios';
+import { SERVER_ADRESS } from '../common';
+
+export async function apiRequestGet(endpoint: string, queryParams: { [key: string]: any }) {
+	const endpointFullString = createApiRequestUrl(endpoint, queryParams);
+	try {
+		const { data } = await axios.get(endpointFullString);
+		if (typeof data !== 'object') {
+			throw new Error('Response was not an object. Response acquired from the server: ' + data);
+		} else if (data.error !== null) {
+			throw new Error(data.error);
+		} else {
+			return data.data;
+		}
+	} catch (error) {
+		console.log(error);
```

Comment:
The error should be somehow rethrown: if it's a string then do `throw new Error(error)`, if it's of type Error then remove the try/catch altogether, if it's some custom axios object then do `throw new Error(<extract message>)`.